kroxylicious-runtime/src/test/java/io/kroxylicious/proxy/internal/net/BootstrapServerSelectionTest.java

```diff
@@ -0,0 +1,93 @@
+package io.kroxylicious.proxy.internal.net;
+
+import java.util.List;
+import java.util.HashSet;
+import java.util.Set;
+
+import org.junit.jupiter.api.Test;
+
+import io.kroxylicious.proxy.service.HostPort;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+/**
+ * Test to verify that bootstrap server selection distributes across multiple servers
+ * instead of always selecting the first one.
+ */
+class BootstrapServerSelectionTest {
+    
+    @Test
+    void shouldSelectDifferentBootstrapServersOverTime() {
+        // Arrange
+        List<HostPort> bootstrapServers = List.of(
+            HostPort.parse("server1:9092"),
+            HostPort.parse("server2:9092"), 
+            HostPort.parse("server3:9092")
+        );
+        
+        EndpointRegistry registry = new EndpointRegistry(new TestNetworkBindingOperationProcessor());
```

Comment:
I **think** I'd use a Mockito mock for NetworkBindingOperationProcessor.  It would verify that it has received the a request for the bind to the expected network address.